fix(rust): Add warning about #[tokio::main] (#7633)
getsentry/sentry-docs: src/platforms/rust/index.mdx

The most convenient way to use this library is the `sentry::init` function, which starts a Sentry client with a default set of integrations, and binds it to the current Hub.

> The `sentry::init` function returns a guard that, when dropped, will flush Events that were not yet sent to the Sentry service. It has a two second deadline for this, so shutdown of applications might slightly delay as a result. Keep the guard around or sending events will not work.

Reviewer: ```suggestion
The `sentry::init` function returns a guard that when dropped, will flush Events that weren't yet sent to Sentry. It has a two-second deadline, so application shutdown may be slightly delayed as a result. Be sure to keep the guard or you won't be able to send events.
```

> **Important:** Note your DSN. The DSN (Data Source Name) tells the SDK where to send events. If you forget it, view Settings -> Projects -> Client Keys (DSN) in sentry.io.

Reviewer: ```suggestion
**Important:** Remember your DSN. The DSN (Data Source Name) tells the SDK where to send events. If you forget it you can find it be going to: Settings -> Projects -> Client Keys (DSN) in sentry.io.
```

> ### Async main function

Reviewer: ```suggestion
### Async Main Function
```

> The Sentry client must be initialized before any async runtime is started or threads are spawned. Unfortunately, this makes it impossible to use macros such as `#[tokio::main]` or `#[actix_web::main]`, because they start the runtime first. Thus, instead of

Reviewer: ```suggestion
The Sentry client must be initialized before any async runtime is started, otherwise threads will be spawned. This means you'll have to avoid using macros such as `#[tokio::main]` or `#[actix_web::main]`, because they start the runtime first. So rather than doing this:
```

Author: Ah, here my formulation was clearly ambiguous. I meant "The Sentry client must be initialized before (any async runtime is started or threads are spawned)".

How does 
> The Sentry client must be initialized before starting an async runtime or spawning threads.

sound?

```rust {filename:main.rs}
// WRONG
#[tokio::main]
async fn main() {
let _guard = sentry::init(("___PUBLIC_DSN___", sentry::ClientOptions {
    release: sentry::release_name!(),
    ..Default::default()
}));

// implementation of main
}
```

> do this:

Reviewer: ```suggestion
Do this instead:
```
